src/game-interface/state-actions.py

```diff
@@ -0,0 +1,143 @@
+# Fireplace-related imports do not appear to be necessary since this function
+# takes a player object
+
+def state_actions(player):
+  # In utils.py, player.hero.power appears to return an object representing
+  # the player's hero power. It does not appear to be explicity declared in
+  # player.py, so this check is similar to that used withing the 
+  # actionable_entities() function
+  if player.hero.power:
+    heropower = player.hero.power
+
+    if heropower.is_usable() and heropower.has_target():
+      # Another attribute defined nowhere else in the API
+      # Unsure what this returns as of yet, storing targets for now
+      hpowactions = heropower.targets
+
+  # Getting the hand directly from the player seems better than passing
+  # in the player's hand into our future function
+  for card in player.hand:
+    if card.is_playable():
+      # This next check appears to be for certain Druid class cards
+      if card.must_choose_one:
+        cardactions = card.choose_cards
+      if card.has_target():
+        cardactions = cardactions + card.targets
```

Comment:
I think this is actually covered by the top level check using _is_playable()_. For the latter part of your questions (spells with no target), this is covered by the following conditional within _is_playable()_:

if _PlayReq.REQ_TARGET_TO_PLAY_ in _self.requirements_

The purpose of the _has_target()_ check in this function is to get the different possibilities for what target a card can be used on. A likely example is that the bot can play a certain spell card (they meet the requirements on the card), but it could be used on multiple targets on the field, so all of the different target possibilities are stored in the cardactions list.